python/lsst/meas/algorithms/stamps.py

```diff
@@ -292,17 +298,21 @@ class StampsBase(abc.ABC, Sequence):
     A butler can be used to read only a part of the stamps,
     specified by a bbox:
 
-    >>> starSubregions = butler.get("brightStarStamps", dataId, parameters={'bbox': bbox})
+    >>> starSubregions = butler.get(
+            "brightStarStamps",
+            dataId,
+            parameters={"bbox": bbox}
+        )
     """
 
-    def __init__(self, stamps, metadata=None, use_mask=True, use_variance=True,
-                 use_archive=False):
-        if not hasattr(stamps, '__iter__'):
-            raise ValueError('The stamps parameter must be iterable.')
+    def __init__(self, stamps, metadata=None, use_mask=True, use_variance=True, use_archive=False):
+        if not hasattr(stamps, "__iter__"):
+            raise ValueError("The stamps parameter must be iterable.")
```

Comment:
Is this check necessary? We don't normally check and raise an error in our code. Moreover, this base class is probably called only within our own code, so we should be able to quickly identify the error if we call it incorrectly.

Reply:
You can probably drop it because the line below `for stamp in stamps` is going to prove whether or not it is iterable -- it clearly has to be. I believe this check was added because there were bugs being triggered by people passing in a single stamp rather than list of stamp. The error message is more explicit  than waiting for the next line to raise a generic error.

Reply:
I've removed this `hasattr` check. I agree, it doesn't quite fit here, and the for loop immediately following will trigger a `TypeError` if a non-iterable type has been passed.